tests/test_dia_object.py

```diff
@@ -20,28 +20,141 @@
+        # Add a flux at some point
 
-    def test_dia_source_append(self):
-        pass
+    return sources
 
-    def compute_light_curve(self):
-        pass
 
-    def compute_summary_statistics(self):
+class TestDIAObject(unittest.TestCase):
+
+    def setUp(self):
+        """ Create the DIAObjct schema we will use throughout these tests.
+        """
+        self.dia_obj_schema = make_minimal_dia_object_schema()
+
+    def tearDown(self):
+        """ Delete the schema when we are done.
+        """
+        del self.dia_obj_schema
+
+    def test_init(self):
+        """ Test DIAObject creation and if we can properly instantiate a
+        DIAObject from an already create dia_object_record.
+        """
+        single_source = create_test_dia_sources(1)
+
+        dia_obj = DIAObject(single_source, None)
+        dia_obj_dup = DIAObject(single_source, dia_obj.dia_object_record)
+
+        self._compare_dia_object_values(dia_obj, 0, 0.0, np.nan)
+        self._compare_dia_object_values(dia_obj_dup, dia_obj.id, 0.0, np.nan)
+
+    def _compare_dia_object_values(self, dia_object, expected_id,
+                                   expected_mean, expected_std):
+        """ Compare values computed in the compute_summary_statistics
+        DIAObject class method with those expected.
+
+        Parameters
+        ----------
+        dia_object : lsst.ap.association.DIAObject
+            Input DIAObect to test
+        expected_id : int
+            Expected id of the DIAObject
+        expected_mean : float
+            Expected mean value of the parameters to be tested.
+        expected_std : float
+            Expected standard diviation of the DIAObject.
+
+        Return
+        ------
+        None
+        """
+
+        for name in self.dia_obj_schema.getNames():
+            if name == 'parent':
+                continue
+            if name == 'id':
+                self.assertEqual(dia_object.get(name), expected_id)
+                continue
+
+            if name[-3:] != 'rms':
+                self.assertAlmostEqual(dia_object.get(name).asDegrees(),
+                                       expected_mean)
+            elif name[-3:] == 'rms' and np.isfinite(expected_std):
+                self.assertAlmostEqual(dia_object.get(name).asDegrees(),
+                                       expected_std)
+            elif name[-3:] == 'rms' and not np.isfinite(expected_std):
+                self.assertFalse(np.isfinite(dia_object.get(name).asDegrees()))
+
+        return None
+
+    def test_update(self):
+        """ If we instantiate the class with a set of DIASources we test to
+        make sure that the DIAObject is intantiated correctly and that we
+        compute the summary statistics in the expected way.
+        """
+        sources = create_test_dia_sources(5)
+        dia_obj = DIAObject(sources, None)
+
+        self._compare_dia_object_values(dia_obj, 0, 2.0, 1.4142135623730951)
```

Comment:
I don't think that actually makes it easier; say you now want to compare a mean flux, you'll still have to go to every instance of `_compare_dia_object_values` and change them. At least make these keyword arguments so they're not just anonymous numbers laying around.